sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -287,20 +288,16 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
 
   override def onTaskFailed(failureReason: TaskFailedReason): Unit = {
     failureReason match {
-      case ef: ExceptionFailure =>
-        val unrecoverableErrors = Seq("cudaErrorIllegalAddress", "cudaErrorLaunchTimeout",
-          "cudaErrorHardwareStackError", "cudaErrorIllegalInstruction",
-          "cudaErrorMisalignedAddress", "cudaErrorInvalidAddressSpace", "cudaErrorInvalidPc",
-          "cudaErrorLaunchFailure", "cudaErrorExternalDevice", "cudaErrorUnknown",
-          "cudaErrorECCUncorrectable")
-        if (unrecoverableErrors.exists(ef.description.contains(_)) ||
-          unrecoverableErrors.exists(ef.toErrorString.contains(_))) {
-          logError("Stopping the Executor based on exception being a fatal CUDA error: " +
-            s"${ef.toErrorString}")
-          System.exit(20)
-        }
+      case ef: ExceptionFailure if RapidsExecutorPlugin.isCudaFatalException(ef) =>
+        logError("Stopping the Executor based on exception being a fatal CUDA error: " +
+          s"${ef.toErrorString}")
+        System.exit(20)
+      case ef: ExceptionFailure if RapidsExecutorPlugin.isCudaException(ef) =>
+        logDebug(s"Executor onTaskFailed because of a non-fatal CUDA error: ${ef.toErrorString}")
+      case ef: ExceptionFailure if RapidsExecutorPlugin.isCudfException(ef) =>
+        logDebug(s"Executor onTaskFailed because of a CUDF error: ${ef.toErrorString}")
```

Comment:
Nit: I think the separate predicate functions are a bit overkill, and this can be simplified a bit by removing the redundant ExceptionFailure checking for each case, e.g.:
```suggestion
      case ef: ExceptionFailure =>
        ef.exception match {
          case _: CudaFatalException =>
            logError("Stopping the Executor based on exception being a fatal CUDA error: " +
              s"${ef.toErrorString}")
            System.exit(20)
          case _: CudaException =>
            logDebug(s"Executor onTaskFailed because of a non-fatal CUDA error: ${ef.toErrorString}")
          case _: CudfException =>
            logDebug(s"Executor onTaskFailed because of a CUDF error: ${ef.toErrorString}")
          case _ =>
            logDebug(s"Executor onTaskFailed: ${ef.toErrorString}")
        }
```